org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/NewCUProposal.java

```diff
@@ -49,9 +49,9 @@
 import org.eclipse.jdt.internal.core.manipulation.util.BasicElementLabels;
 import org.eclipse.jdt.internal.corext.dom.ASTNodes;
 import org.eclipse.jdt.internal.corext.dom.IASTSharedValues;
+import org.eclipse.jdt.internal.corext.refactoring.nls.changes.CreateFileChange;
```

Comment:
This small little mistake introduced a regression. Thanks to @fbricon for catching this. Currently creating compilation units (classes/interfaces/enums) from unresolved symbols fails to create the file. You can see the missing

```
"options": {
"overwrite": false,
"ignoreIfExists": true
},
"kind": "create"
```
when comparing the data.

When I planned this refactoring, I wanted to use `CreateFileChange` from `org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/nls/changes/CreateFileChange.java` and remove our copy. However, I noticed that a key difference. Our copy opened up `getPath()` whereas it's restricted in the upstream version.

```
-   protected IPath getPath() {
+   public IPath getPath() {
        return fPath;
    }
```

Rather than wait for it to change, I decided to just undo the change and continue using out copy of `CreateFileChange`. I must have left this import in, and because it resolves, I never saw any indication of a problem.

The result is that we use [`org.eclipse.jdt.internal.corext.refactoring.nls.changes.CreateFileChange`](https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/NewCUProposal.java#L277) when creating the file change, but our [`ChangeUtil`](https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ChangeUtil.java#L150) code does an instance of check on our `org.eclipse.jdt.ls.core.internal.corext.refactoring.nls.changes.CreateFileChange` so we fail to correctly recognize this case.

Just changing this line fixes things, so I'll post a change, and hopefully there's a way to test this.